src/solana/SolanaEngine.ts

```diff
@@ -775,13 +775,15 @@ export class SolanaEngine extends CurrencyEngine<
     })
     const versionedTx = new VersionedTransaction(versionedMessage)
 
-    const cleanSpendInfoOtherParams =
-      asSolanaSpendInfoOtherParams(spendInfoOtherParams)
+    let unsignedTx = versionedTx.serialize()
+    if (spendInfoOtherParams?.unsignedTx != null) {
+      const cleanSpendInfoOtherParams =
+        asSolanaTxOtherParams(spendInfoOtherParams)
+      unsignedTx = cleanSpendInfoOtherParams.unsignedTx
+    }
```

Comment:
We need to do this:

```ts
const asSolanaSpendInfoOtherParams = asObject({
  unsignedTx: asOptional(asBase64)
})

export const asSolanaTxOtherParams = asObject({
  unsignedTx: asBase64
})
```

Then use `asSolanaSpendInfoOtherParams` here to clean the input from the GUI. Since the `unsignedTx` is optional, we can safely clean the input before we open the field. The `asSolanaTxOtherParams` has the `unsignedTx` as mandatory, since we need that to sign and send.